test/Botwin.Tests/ResponseFromStreamTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿namespace Botwin.Tests
+{
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.AspNetCore.TestHost;
+    using Xunit;
+
+    public class ResponseFromStreamTests
+    {
+        private TestServer server;
+
+        private HttpClient httpClient;
+
+        public ResponseFromStreamTests()
+        {
+            this.server = new TestServer(new WebHostBuilder()
+                .ConfigureServices(x => { x.AddBotwin(); })
+                .Configure(x => x.UseBotwin())
+            );
+            this.httpClient = this.server.CreateClient();
+        }
+
+        [Fact]
+        public async Task Should_set_content_type_body_acceptrange_header_content_disposition()
+        {
+            //Given & When
+            var response = await this.httpClient.GetAsync("/downloadwithcd");
+            var body = await response.Content.ReadAsStringAsync();
+            var filename = response.Content.Headers.ContentDisposition.FileName;
+
+            //Then
+            Assert.Equal("application/csv", response.Content.Headers.ContentType.MediaType);
+            Assert.Equal("hi", body);
+            Assert.Equal("bytes", response.Headers.AcceptRanges.FirstOrDefault());
```

Comment:
nit: Consistency: `response.Headers` vs. `response.Content.Headers`

Reply:
Blame ASPNET team, some are exposed on Headers whilst others are on response.Content.Headers. Weird

Reply:
`HttpContentHeaders` and `HttpResponseHeaders`… meh >_<"